src/System.IO.FileSystem/tests/FileStream/ctor_str_fm_fa_fs.cs

```diff
@@ -117,5 +118,26 @@ public void FileShareOpenOrCreate()
                 }
             }
         }
+
+        [Theory]
+        [InlineData(FileMode.Create)]
+        [InlineData(FileMode.Truncate)]
+        public void NoTruncateOnFileShareViolation(FileMode fileMode)
+        {
+            string fileName = GetTestFilePath();
+
+            using (FileStream fs = CreateFileStream(fileName, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
+            {
+                fs.Write(new byte[] { 42 }, 0, 1);
+                fs.Flush();
+                FSAssert.ThrowsSharingViolation(() => CreateFileStream(fileName, fileMode, FileAccess.Write, FileShare.None).Dispose());
```

Comment:
Nit: the Dispose shouldn't be necessary

Reply:
But if it doesn't throw then we leave the file open until a GC occurs, does that not matter?

Reply:
> But if it doesn't throw

That's true, but if it doesn't throw, we have bigger problems :)  I'm ok with it either way, but in general we haven't worried about such things elsewhere.